### PR TITLE
add and delete service token creator role

### PR DESCRIPTION
**Pull request by MengZhuNAV**
only on the service account itself

### CODE REVIEW

File: pkg/chart/gcp.go [hunk 1]
```diff
@@ -443,3 +443,55 @@ func roleBindingExistsInGCP(ctx context.Context, gcpProject, teamID, role string
 	roles := strings.Split(stdOut.String(), "\n")
 	return slices.Contains(roles, role), nil
 }
+
+func grantSATokenCreatorRole(ctx context.Context, teamID, gcpProject string) error {
+	role := "roles/iam.serviceAccountTokenCreator"
+
+	var sa = fmt.Sprintf("%v@%v.iam.gserviceaccount.com", teamID, gcpProject)
```
Comment (erikvatt) [inline pkg/chart/gcp.go:450]: I'd use := variable declaration, same as role above, but that's up to you.

File: pkg/chart/gcp.go [hunk 2]
```diff
@@ -443,3 +443,55 @@ func roleBindingExistsInGCP(ctx context.Context, gcpProject, teamID, role string
...
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("%v\nstderr: %v", err, stdErr.String())
+	}
+	return nil
+}
+
+func deleteTokenCreatorRoleOnSA(ctx context.Context, teamID, gcpProject string) error {
+	role := "roles/iam.serviceAccountTokenCreator"
+
+	var sa = fmt.Sprintf("%v@%v.iam.gserviceaccount.com", teamID, gcpProject)
```
Comment (erikvatt) [inline pkg/chart/gcp.go:476]: Same as above comment

**Review by erikvatt — APPROVED** [review]

File: pkg/chart/gcp.go (hunk 1 above)
Comment (MengZhuNAV) [reply pkg/chart/gcp.go]: Ok!

File: pkg/chart/gcp.go [hunk 3]
```diff
@@ -443,3 +443,55 @@ func roleBindingExistsInGCP(ctx context.Context, gcpProject, teamID, role string
...
+	sa := fmt.Sprintf("%v@%v.iam.gserviceaccount.com", teamID, gcpProject)
+
+	cmd := exec.CommandContext(ctx,
+		"gcloud",
+		"iam",
+		"service-accounts",
+		"add-iam-policy-binding",
+		sa,
+		fmt.Sprintf("--role=%v", role),
+		fmt.Sprintf("--member=serviceAccount:%v", sa),
```
Comment (Kyrremann) [inline pkg/chart/gcp.go:459]: Do we really need `--member` when we specify `sa` two lines above? Same question for the delete-function.

Comment (MengZhuNAV) [reply pkg/chart/gcp.go:459]: Yes, we need to restrict the scope to the service account, not the project wild

**Pull request merged**